base/task/sequenced_task_runner.cc

```diff
@@ -104,11 +104,20 @@ bool SequencedTaskRunner::PostDelayedTaskAt(
                              : delayed_run_time - TimeTicks::Now());
 }
 
+#if defined(STARBOARD)
+// static
+scoped_refptr<SequencedTaskRunner>*
+    SequencedTaskRunner::null_sequenced_task_runner_(
+        new scoped_refptr<SequencedTaskRunner>);
+#endif
+
 // static
 const scoped_refptr<SequencedTaskRunner>&
 SequencedTaskRunner::GetCurrentDefault() {
 #if defined(STARBOARD)
   auto current_default_handle = GetCurrentDefaultHandle();
+  return (!current_default_handle ? *null_sequenced_task_runner_
```

Comment:
minor quibble, but aren't you taking the `CHECK(..)` below out of the code here ? I mean the codebase at this point in the branch is quite stable and it's unlikely we'll see this tripping, because there are no new callers, but nevertheless ... 

Maybe don't change this part at all and don't set up null runner, and just assert/CHECK in `PostTaskAndReply` that we did get the runner.

Reply:
I understand that you are turning the CHECK to dead code with the return that wasn't here before - i'm asking if this is a good thing to do ?

Does it actually trip on the CHECK when you remove the early return here ? If it doesn't, then the return seems like a no-op.

Reply:
The CHECK /here/ is turned into dead code exactly because the function is now allowed to be called when there is no current default task runner, instead of relying on callers using a costly additional call to `HasCurrentDefault` to gate whether or not to call `GetCurrentDefault`.

And yes, the check trips when `GetCurrentDefault` is called without a current default task runner, and there is a unit test that posts a task when there isn't a default task runner, which relies on the `HasCurrentDefault` to be used to avoid calling `GetCurrentDefault`.